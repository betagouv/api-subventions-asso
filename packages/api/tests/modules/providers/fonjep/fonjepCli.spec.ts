import path from "path";
import db from "../../../../src/shared/MongoConnection";
import FonjepCli from "../../../../src/interfaces/cli/Fonjep.cli";
import fonjepVersementPort from "../../../../src/dataProviders/db/providers/fonjep/fonjep.versements.port";
import { ObjectId } from "mongodb";
import fonjepTiersPort from "../../../../src/dataProviders/db/providers/fonjep/fonjep.tiers.port";
import fonjepPostesPort from "../../../../src/dataProviders/db/providers/fonjep/fonjep.postes.port";
import fonjepTypePostePort from "../../../../src/dataProviders/db/providers/fonjep/fonjep.typePoste.port";
import fonjepDispositifPort from "../../../../src/dataProviders/db/providers/fonjep/fonjep.dispositif.port";
import dataBretagnePort from "../../../../src/dataProviders/api/dataBretagne/dataBretagne.port";
import { DATA_BRETAGNE_DTOS, PROGRAMS } from "../../../__fixtures__/paymentsFlat.fixture";
import stateBudgetProgramPort from "../../../../src/dataProviders/db/state-budget-program/stateBudgetProgram.port";
import paymentFlatPort from "../../../../src/dataProviders/db/paymentFlat/paymentFlat.port";
import applicationFlatPort from "../../../../src/dataProviders/db/applicationFlat/applicationFlat.port";

const FILEPATH = path.resolve(__dirname, "./__fixtures__/fonjep-new.xlsx");
const EXPORT_DATE = new Date("2022-03-03").toISOString();

describe("FonjepCli", () => {
    let cli: FonjepCli;

    beforeEach(async () => {
        // mock API call to DataBretagne
        jest.spyOn(dataBretagnePort, "login").mockImplementation(jest.fn());
        jest.spyOn(dataBretagnePort, "getCollection").mockImplementation(collection => DATA_BRETAGNE_DTOS[collection]);
        await stateBudgetProgramPort.replace(PROGRAMS);
    });

    describe("parse()", () => {
        beforeAll(async () => {
            cli = new FonjepCli();
        });

        describe("test with all information arround one code poste", () => {
            it("adds applications flat", async () => {
                await cli.parse(path.resolve(__dirname, "./__fixtures__/fonjep-one-code-full.xlsx"), "2025-09-30");
                const actual = await applicationFlatPort.findAll();
                expect(actual).toMatchSnapshot();
            });
            it("adds payments flat", async () => {
                await cli.parse(path.resolve(__dirname, "./__fixtures__/fonjep-one-code-full.xlsx"), "2025-09-30");
                const actual = await paymentFlatPort.findAll();
                expect(actual).toMatchSnapshot();
            });
        });

        it("should remove temporary collections", async () => {
            await cli.parse(FILEPATH, EXPORT_DATE);
            const collections = await db.listCollections().toArray();
            const actual = collections.find(col =>
                [
                    "fonjepDispositif-tmp-collection",
                    "fonjepPoste-tmp-collection",
                    "FonjepTiers-tmp-collection",
                    "FonjepTypePoste-tmp-collection",
                    "FonjepVersement-tmp-collection",
                ].includes(col.name),
            );
            expect(actual).toBeUndefined();
        });

        it("should create or replace Versement collection", async () => {
            await cli.parse(FILEPATH, EXPORT_DATE);
            const actualVersement = await fonjepVersementPort.findAll();
            const expected = actualVersement.map(() => ({
                _id: expect.any(ObjectId),
            }));
            expect(actualVersement).toMatchSnapshot(expected);
        });

        it("should create or replace Tiers collection", async () => {
            await cli.parse(FILEPATH, EXPORT_DATE);
            const actualTiers = await fonjepTiersPort.findAll();
            const expected = actualTiers.map(() => ({
                _id: expect.any(ObjectId),
            }));
            expect(actualTiers).toMatchSnapshot(expected);
        });

        it("should create or replace Poste collection", async () => {
            await cli.parse(FILEPATH, EXPORT_DATE);
            const actualPoste = await fonjepPostesPort.findAll();
            const expected = actualPoste.map(() => ({
                _id: expect.any(ObjectId),
            }));
            expect(actualPoste).toMatchSnapshot(expected);
        });

        it("should create or remplace TypePoste collection", async () => {
            await cli.parse(FILEPATH, EXPORT_DATE);
            const actualTypePoste = await fonjepTypePostePort.findAll();
            const expected = actualTypePoste.map(() => ({
                _id: expect.any(ObjectId),
            }));
            expect(actualTypePoste).toMatchSnapshot(expected);
        });

        it("should create or replace Dispositif collection", async () => {
            await cli.parse(FILEPATH, EXPORT_DATE);
            const actualDispositif = await fonjepDispositifPort.findAll();
            const expected = actualDispositif.map(() => ({
                _id: expect.any(ObjectId),
            }));
            expect(actualDispositif).toMatchSnapshot(expected);
        });

        it("should add FonjepPaymentFlat", async () => {
            await cli.parse(FILEPATH, EXPORT_DATE);
            const paymentsFlat = await paymentFlatPort.findAll();
            expect(paymentsFlat.map(flat => ({ ...flat, _id: expect.any(String) }))).toMatchSnapshot();
        });

        it("should add FonjepApplicationFlat", async () => {
            await cli.parse(FILEPATH, EXPORT_DATE);
            const applications = await applicationFlatPort.findAll();
            expect(
                applications.map(flat => ({ ...flat, _id: expect.any(String), updateDate: expect.any(Date) })),
            ).toMatchSnapshot();
        });
    });
});
