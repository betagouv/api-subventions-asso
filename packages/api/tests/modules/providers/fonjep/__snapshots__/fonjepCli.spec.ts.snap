// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FonjepCli parse() should add FonjepPaymentFlat 1`] = `
[
  {
    "_id": Any<String>,
    "action": "N/A",
    "activite": "N/A",
    "attachementComptable": "N/A",
    "codeAction": "N/A",
    "codeActivite": "N/A",
    "codeCentreFinancier": "N/A",
    "dateOperation": 2017-01-26T00:00:00.000Z,
    "ej": "N/A",
    "exerciceBudgetaire": 2017,
    "idEntrepriseBeneficiaire": "000000000",
    "idEtablissementBeneficiaire": "00000000000000",
    "idVersement": "J10540-2015-12-31-2017-00000000000000",
    "libelleCentreFinancier": "N/A",
    "ministere": "nom_ministere_1",
    "mission": "mission_1",
    "montant": 1777,
    "numeroProgramme": 163,
    "programme": "programme_1",
    "provider": "fonjep",
    "regionAttachementComptable": "N/A",
    "sigleMinistere": "sigle_ministere_1",
    "typeIdEntrepriseBeneficiaire": "siren",
    "typeIdEtablissementBeneficiaire": "siret",
    "uniqueId": "fonjep-J10540-2015-12-31-2017-00000000000000-163-N/A-N/A-2017-01-26-N/A-N/A",
  },
  {
    "_id": Any<String>,
    "action": "N/A",
    "activite": "N/A",
    "attachementComptable": "N/A",
    "codeAction": "N/A",
    "codeActivite": "N/A",
    "codeCentreFinancier": "N/A",
    "dateOperation": 2017-04-27T00:00:00.000Z,
    "ej": "N/A",
    "exerciceBudgetaire": 2017,
    "idEntrepriseBeneficiaire": "000000000",
    "idEtablissementBeneficiaire": "00000000000000",
    "idVersement": "J10540-2015-12-31-2017-00000000000000",
    "libelleCentreFinancier": "N/A",
    "ministere": "nom_ministere_1",
    "mission": "mission_1",
    "montant": 1777,
    "numeroProgramme": 163,
    "programme": "programme_1",
    "provider": "fonjep",
    "regionAttachementComptable": "N/A",
    "sigleMinistere": "sigle_ministere_1",
    "typeIdEntrepriseBeneficiaire": "siren",
    "typeIdEtablissementBeneficiaire": "siret",
    "uniqueId": "fonjep-J10540-2015-12-31-2017-00000000000000-163-N/A-N/A-2017-04-27-N/A-N/A",
  },
  {
    "_id": Any<String>,
    "action": "N/A",
    "activite": "N/A",
    "attachementComptable": "N/A",
    "codeAction": "N/A",
    "codeActivite": "N/A",
    "codeCentreFinancier": "N/A",
    "dateOperation": 2017-07-11T00:00:00.000Z,
    "ej": "N/A",
    "exerciceBudgetaire": 2017,
    "idEntrepriseBeneficiaire": "000000000",
    "idEtablissementBeneficiaire": "00000000000000",
    "idVersement": "J10540-2015-12-31-2017-00000000000000",
    "libelleCentreFinancier": "N/A",
    "ministere": "nom_ministere_1",
    "mission": "mission_1",
    "montant": 1777,
    "numeroProgramme": 163,
    "programme": "programme_1",
    "provider": "fonjep",
    "regionAttachementComptable": "N/A",
    "sigleMinistere": "sigle_ministere_1",
    "typeIdEntrepriseBeneficiaire": "siren",
    "typeIdEtablissementBeneficiaire": "siret",
    "uniqueId": "fonjep-J10540-2015-12-31-2017-00000000000000-163-N/A-N/A-2017-07-11-N/A-N/A",
  },
  {
    "_id": Any<String>,
    "action": "N/A",
    "activite": "N/A",
    "attachementComptable": "N/A",
    "codeAction": "N/A",
    "codeActivite": "N/A",
    "codeCentreFinancier": "N/A",
    "dateOperation": 2017-10-19T00:00:00.000Z,
    "ej": "N/A",
    "exerciceBudgetaire": 2017,
    "idEntrepriseBeneficiaire": "000000000",
    "idEtablissementBeneficiaire": "00000000000000",
    "idVersement": "J10540-2015-12-31-2017-00000000000000",
    "libelleCentreFinancier": "N/A",
    "ministere": "nom_ministere_1",
    "mission": "mission_1",
    "montant": 1776,
    "numeroProgramme": 163,
    "programme": "programme_1",
    "provider": "fonjep",
    "regionAttachementComptable": "N/A",
    "sigleMinistere": "sigle_ministere_1",
    "typeIdEntrepriseBeneficiaire": "siren",
    "typeIdEtablissementBeneficiaire": "siret",
    "uniqueId": "fonjep-J10540-2015-12-31-2017-00000000000000-163-N/A-N/A-2017-10-19-N/A-N/A",
  },
  {
    "_id": Any<String>,
    "action": "N/A",
    "activite": "N/A",
    "attachementComptable": "N/A",
    "codeAction": "N/A",
    "codeActivite": "N/A",
    "codeCentreFinancier": "N/A",
    "dateOperation": 2017-01-26T00:00:00.000Z,
    "ej": "N/A",
    "exerciceBudgetaire": 2017,
    "idEntrepriseBeneficiaire": "000000000",
    "idEtablissementBeneficiaire": "00000000000000",
    "idVersement": "J23425-2015-12-31-2017-00000000000000",
    "libelleCentreFinancier": "N/A",
    "ministere": "nom_ministere_2",
    "mission": "mission_2",
    "montant": 1777,
    "numeroProgramme": 361,
    "programme": "programme_2",
    "provider": "fonjep",
    "regionAttachementComptable": "N/A",
    "sigleMinistere": "sigle_ministere_2",
    "typeIdEntrepriseBeneficiaire": "siren",
    "typeIdEtablissementBeneficiaire": "siret",
    "uniqueId": "fonjep-J23425-2015-12-31-2017-00000000000000-361-N/A-N/A-2017-01-26-N/A-N/A",
  },
  {
    "_id": Any<String>,
    "action": "N/A",
    "activite": "N/A",
    "attachementComptable": "N/A",
    "codeAction": "N/A",
    "codeActivite": "N/A",
    "codeCentreFinancier": "N/A",
    "dateOperation": 2017-04-27T00:00:00.000Z,
    "ej": "N/A",
    "exerciceBudgetaire": 2017,
    "idEntrepriseBeneficiaire": "000000000",
    "idEtablissementBeneficiaire": "00000000000000",
    "idVersement": "J23425-2015-12-31-2017-00000000000000",
    "libelleCentreFinancier": "N/A",
    "ministere": "nom_ministere_2",
    "mission": "mission_2",
    "montant": 1777,
    "numeroProgramme": 361,
    "programme": "programme_2",
    "provider": "fonjep",
    "regionAttachementComptable": "N/A",
    "sigleMinistere": "sigle_ministere_2",
    "typeIdEntrepriseBeneficiaire": "siren",
    "typeIdEtablissementBeneficiaire": "siret",
    "uniqueId": "fonjep-J23425-2015-12-31-2017-00000000000000-361-N/A-N/A-2017-04-27-N/A-N/A",
  },
  {
    "_id": Any<String>,
    "action": "N/A",
    "activite": "N/A",
    "attachementComptable": "N/A",
    "codeAction": "N/A",
    "codeActivite": "N/A",
    "codeCentreFinancier": "N/A",
    "dateOperation": 2017-07-11T00:00:00.000Z,
    "ej": "N/A",
    "exerciceBudgetaire": 2017,
    "idEntrepriseBeneficiaire": "000000000",
    "idEtablissementBeneficiaire": "00000000000000",
    "idVersement": "J23425-2015-12-31-2017-00000000000000",
    "libelleCentreFinancier": "N/A",
    "ministere": "nom_ministere_2",
    "mission": "mission_2",
    "montant": 1777,
    "numeroProgramme": 361,
    "programme": "programme_2",
    "provider": "fonjep",
    "regionAttachementComptable": "N/A",
    "sigleMinistere": "sigle_ministere_2",
    "typeIdEntrepriseBeneficiaire": "siren",
    "typeIdEtablissementBeneficiaire": "siret",
    "uniqueId": "fonjep-J23425-2015-12-31-2017-00000000000000-361-N/A-N/A-2017-07-11-N/A-N/A",
  },
  {
    "_id": Any<String>,
    "action": "N/A",
    "activite": "N/A",
    "attachementComptable": "N/A",
    "codeAction": "N/A",
    "codeActivite": "N/A",
    "codeCentreFinancier": "N/A",
    "dateOperation": 2018-09-06T00:00:00.000Z,
    "ej": "N/A",
    "exerciceBudgetaire": 2017,
    "idEntrepriseBeneficiaire": "000000000",
    "idEtablissementBeneficiaire": "00000000000000",
    "idVersement": "J23425-2015-12-31-2017-00000000000000",
    "libelleCentreFinancier": "N/A",
    "ministere": "nom_ministere_2",
    "mission": "mission_2",
    "montant": 1776,
    "numeroProgramme": 361,
    "programme": "programme_2",
    "provider": "fonjep",
    "regionAttachementComptable": "N/A",
    "sigleMinistere": "sigle_ministere_2",
    "typeIdEntrepriseBeneficiaire": "siren",
    "typeIdEtablissementBeneficiaire": "siret",
    "uniqueId": "fonjep-J23425-2015-12-31-2017-00000000000000-361-N/A-N/A-2018-09-06-N/A-N/A",
  },
  {
    "_id": Any<String>,
    "action": "N/A",
    "activite": "N/A",
    "attachementComptable": "N/A",
    "codeAction": "N/A",
    "codeActivite": "N/A",
    "codeCentreFinancier": "N/A",
    "dateOperation": 2017-01-26T00:00:00.000Z,
    "ej": "N/A",
    "exerciceBudgetaire": 2022,
    "idEntrepriseBeneficiaire": "000000000",
    "idEtablissementBeneficiaire": "00000000000000",
    "idVersement": "J02924-2021-12-31-2022-00000000000000",
    "libelleCentreFinancier": "N/A",
    "ministere": "nom_ministere_1",
    "mission": "mission_1",
    "montant": 1791,
    "numeroProgramme": 163,
    "programme": "programme_1",
    "provider": "fonjep",
    "regionAttachementComptable": "N/A",
    "sigleMinistere": "sigle_ministere_1",
    "typeIdEntrepriseBeneficiaire": "siren",
    "typeIdEtablissementBeneficiaire": "siret",
    "uniqueId": "fonjep-J02924-2021-12-31-2022-00000000000000-163-N/A-N/A-2017-01-26-N/A-N/A",
  },
  {
    "_id": Any<String>,
    "action": "N/A",
    "activite": "N/A",
    "attachementComptable": "N/A",
    "codeAction": "N/A",
    "codeActivite": "N/A",
    "codeCentreFinancier": "N/A",
    "dateOperation": 2017-04-27T00:00:00.000Z,
    "ej": "N/A",
    "exerciceBudgetaire": 2022,
    "idEntrepriseBeneficiaire": "000000000",
    "idEtablissementBeneficiaire": "00000000000000",
    "idVersement": "J02924-2021-12-31-2022-00000000000000",
    "libelleCentreFinancier": "N/A",
    "ministere": "nom_ministere_1",
    "mission": "mission_1",
    "montant": 1791,
    "numeroProgramme": 163,
    "programme": "programme_1",
    "provider": "fonjep",
    "regionAttachementComptable": "N/A",
    "sigleMinistere": "sigle_ministere_1",
    "typeIdEntrepriseBeneficiaire": "siren",
    "typeIdEtablissementBeneficiaire": "siret",
    "uniqueId": "fonjep-J02924-2021-12-31-2022-00000000000000-163-N/A-N/A-2017-04-27-N/A-N/A",
  },
  {
    "_id": Any<String>,
    "action": "N/A",
    "activite": "N/A",
    "attachementComptable": "N/A",
    "codeAction": "N/A",
    "codeActivite": "N/A",
    "codeCentreFinancier": "N/A",
    "dateOperation": 2017-07-11T00:00:00.000Z,
    "ej": "N/A",
    "exerciceBudgetaire": 2022,
    "idEntrepriseBeneficiaire": "000000000",
    "idEtablissementBeneficiaire": "00000000000000",
    "idVersement": "J02924-2021-12-31-2022-00000000000000",
    "libelleCentreFinancier": "N/A",
    "ministere": "nom_ministere_1",
    "mission": "mission_1",
    "montant": 1791,
    "numeroProgramme": 163,
    "programme": "programme_1",
    "provider": "fonjep",
    "regionAttachementComptable": "N/A",
    "sigleMinistere": "sigle_ministere_1",
    "typeIdEntrepriseBeneficiaire": "siren",
    "typeIdEtablissementBeneficiaire": "siret",
    "uniqueId": "fonjep-J02924-2021-12-31-2022-00000000000000-163-N/A-N/A-2017-07-11-N/A-N/A",
  },
  {
    "_id": Any<String>,
    "action": "N/A",
    "activite": "N/A",
    "attachementComptable": "N/A",
    "codeAction": "N/A",
    "codeActivite": "N/A",
    "codeCentreFinancier": "N/A",
    "dateOperation": 2017-10-19T00:00:00.000Z,
    "ej": "N/A",
    "exerciceBudgetaire": 2022,
    "idEntrepriseBeneficiaire": "000000000",
    "idEtablissementBeneficiaire": "00000000000000",
    "idVersement": "J02924-2021-12-31-2022-00000000000000",
    "libelleCentreFinancier": "N/A",
    "ministere": "nom_ministere_1",
    "mission": "mission_1",
    "montant": 1791,
    "numeroProgramme": 163,
    "programme": "programme_1",
    "provider": "fonjep",
    "regionAttachementComptable": "N/A",
    "sigleMinistere": "sigle_ministere_1",
    "typeIdEntrepriseBeneficiaire": "siren",
    "typeIdEtablissementBeneficiaire": "siret",
    "uniqueId": "fonjep-J02924-2021-12-31-2022-00000000000000-163-N/A-N/A-2017-10-19-N/A-N/A",
  },
]
`;

exports[`FonjepCli parse() should create or remplace TypePoste collection 1`] = `
[
  {
    "_id": Any<ObjectId>,
    "code": "DOUBLE",
    "libelle": "UNITE DE COMPTE DOUBLEE",
  },
  {
    "_id": Any<ObjectId>,
    "code": "FONJEP",
    "libelle": "Poste FONJEP",
  },
  {
    "_id": Any<ObjectId>,
    "code": "PSTEMP",
    "libelle": "POSTES EMPOI FONJEP (100 %)",
  },
  {
    "_id": Any<ObjectId>,
    "code": "FONBIS",
    "libelle": "PROLONGATION PAS DE FRAIS",
  },
]
`;

exports[`FonjepCli parse() should create or replace Dispositif collection 1`] = `
[
  {
    "_id": Any<ObjectId>,
    "financeurCode": "10004",
    "id": 1,
    "libelle": "FONJEP Jeunesse éducation populaire",
  },
  {
    "_id": Any<ObjectId>,
    "financeurCode": "10005",
    "id": 2,
    "libelle": "FONJEP Cohésion sociale",
  },
  {
    "_id": Any<ObjectId>,
    "financeurCode": "10008",
    "id": 3,
    "libelle": "FONJEP Politique de la ville",
  },
  {
    "_id": Any<ObjectId>,
    "financeurCode": "10009",
    "id": 4,
    "libelle": "FONJEP Guid'Asso",
  },
  {
    "_id": Any<ObjectId>,
    "financeurCode": "10010",
    "id": 5,
    "libelle": "FONJEP Éducation à la citoyenneté et à la solidarité internationale (ECSI)",
  },
  {
    "_id": Any<ObjectId>,
    "financeurCode": "10012",
    "id": 6,
    "libelle": "FONJEP Culture",
  },
  {
    "_id": Any<ObjectId>,
    "financeurCode": "10016",
    "id": 7,
    "libelle": "FONJEP Centre de ressources et d'information des bénévoles",
  },
  {
    "_id": Any<ObjectId>,
    "financeurCode": "10017",
    "id": 8,
    "libelle": "FONJEP Jeunes",
  },
]
`;

exports[`FonjepCli parse() should create or replace Poste collection 1`] = `
[
  {
    "_id": Any<ObjectId>,
    "annee": 2017,
    "associationBeneficiaireCode": 1,
    "associationImplantationCode": 1,
    "code": "J10540",
    "dateFinTriennalite": 2018-12-31T00:00:00.000Z,
    "dispositifId": "1",
    "doublementUniteCompte": "Non",
    "financeurAttributeurCode": 1,
    "financeurPrincipalCode": 10004,
    "montantSubvention": 6666,
    "pleinTemps": "Oui",
    "pstRaisonStatutLibelle": "Reconduction",
    "pstStatutPosteLibelle": "Attribué",
    "pstTypePosteCode": "FONJEP",
  },
  {
    "_id": Any<ObjectId>,
    "annee": 2022,
    "associationBeneficiaireCode": 2,
    "associationImplantationCode": 2,
    "code": "J02924",
    "dateFinTriennalite": 2024-12-31T00:00:00.000Z,
    "dispositifId": 3,
    "doublementUniteCompte": "Non",
    "financeurAttributeurCode": 2,
    "financeurPrincipalCode": 10005,
    "montantSubvention": 7164,
    "pleinTemps": "Oui",
    "pstRaisonStatutLibelle": "Création",
    "pstStatutPosteLibelle": "Attribué",
    "pstTypePosteCode": "FONJEP",
  },
  {
    "_id": Any<ObjectId>,
    "annee": 2017,
    "associationBeneficiaireCode": 3,
    "associationImplantationCode": 3,
    "code": "J23425",
    "dateFinTriennalite": 2018-12-31T00:00:00.000Z,
    "dispositifId": 2,
    "doublementUniteCompte": "Non",
    "financeurAttributeurCode": 3,
    "financeurPrincipalCode": 10012,
    "montantSubvention": 5142,
    "pleinTemps": "Oui",
    "pstRaisonStatutLibelle": "Création",
    "pstStatutPosteLibelle": "Attribué",
    "pstTypePosteCode": "FONJEP",
  },
]
`;

exports[`FonjepCli parse() should create or replace Tiers collection 1`] = `
[
  {
    "_id": Any<ObjectId>,
    "code": 1,
    "codePostal": "00000",
    "contactEmail": "exemple@beta.gouv.fr",
    "estAssociation": "Oui",
    "estCoFinanceurPostes": "Non",
    "estFinanceurPostes": "Non",
    "raisonSociale": "CENTRE D'INFO ****",
    "siretOuRidet": "00000000000000",
    "ville": "Paris",
  },
  {
    "_id": Any<ObjectId>,
    "code": 2,
    "codePostal": "00000",
    "contactEmail": "exemple@beta.gouv.fr",
    "estAssociation": "Oui",
    "estCoFinanceurPostes": "Non",
    "estFinanceurPostes": "Non",
    "raisonSociale": "CHEMINEE CLARKS",
    "siretOuRidet": "00000000000000",
    "ville": "Paris",
  },
  {
    "_id": Any<ObjectId>,
    "code": 3,
    "codePostal": "00000",
    "contactEmail": "exemple@beta.gouv.fr",
    "estAssociation": "Oui",
    "estCoFinanceurPostes": "Non",
    "estFinanceurPostes": "Non",
    "raisonSociale": "DES IDEES PLEIN LA TERRE",
    "siretOuRidet": "00000000000000",
    "ville": "Paris",
  },
]
`;

exports[`FonjepCli parse() should create or replace versement collection 1`] = `
[
  {
    "_id": Any<ObjectId>,
    "dateVersement": 2017-01-26T00:00:00.000Z,
    "montantAPayer": 1777,
    "montantPaye": 1777,
    "periodeDebut": 2017-01-01T00:00:00.000Z,
    "periodeFin": 2017-03-31T00:00:00.000Z,
    "posteCode": "J10540",
  },
  {
    "_id": Any<ObjectId>,
    "dateVersement": 2017-04-27T00:00:00.000Z,
    "montantAPayer": 1777,
    "montantPaye": 1777,
    "periodeDebut": 2017-04-01T00:00:00.000Z,
    "periodeFin": 2017-06-30T00:00:00.000Z,
    "posteCode": "J10540",
  },
  {
    "_id": Any<ObjectId>,
    "dateVersement": 2017-07-11T00:00:00.000Z,
    "montantAPayer": 1777,
    "montantPaye": 1777,
    "periodeDebut": 2017-07-01T00:00:00.000Z,
    "periodeFin": 2017-09-30T00:00:00.000Z,
    "posteCode": "J10540",
  },
  {
    "_id": Any<ObjectId>,
    "dateVersement": 2017-10-19T00:00:00.000Z,
    "montantAPayer": 1776,
    "montantPaye": 1776,
    "periodeDebut": 2017-10-01T00:00:00.000Z,
    "periodeFin": 2017-12-31T00:00:00.000Z,
    "posteCode": "J10540",
  },
  {
    "_id": Any<ObjectId>,
    "dateVersement": 2017-01-26T00:00:00.000Z,
    "montantAPayer": 1777,
    "montantPaye": 1777,
    "periodeDebut": 2017-01-01T00:00:00.000Z,
    "periodeFin": 2017-03-31T00:00:00.000Z,
    "posteCode": "J23425",
  },
  {
    "_id": Any<ObjectId>,
    "dateVersement": 2017-04-27T00:00:00.000Z,
    "montantAPayer": 1777,
    "montantPaye": 1777,
    "periodeDebut": 2017-04-01T00:00:00.000Z,
    "periodeFin": 2017-06-30T00:00:00.000Z,
    "posteCode": "J23425",
  },
  {
    "_id": Any<ObjectId>,
    "dateVersement": 2017-07-11T00:00:00.000Z,
    "montantAPayer": 1777,
    "montantPaye": 1777,
    "periodeDebut": 2017-07-01T00:00:00.000Z,
    "periodeFin": 2017-09-30T00:00:00.000Z,
    "posteCode": "J23425",
  },
  {
    "_id": Any<ObjectId>,
    "dateVersement": 2018-09-06T00:00:00.000Z,
    "montantAPayer": 1776,
    "montantPaye": 1776,
    "periodeDebut": 2017-10-01T00:00:00.000Z,
    "periodeFin": 2017-12-31T00:00:00.000Z,
    "posteCode": "J23425",
  },
  {
    "_id": Any<ObjectId>,
    "dateVersement": 2017-01-26T00:00:00.000Z,
    "montantAPayer": 1791,
    "montantPaye": 1791,
    "periodeDebut": 2017-01-01T00:00:00.000Z,
    "periodeFin": 2017-03-31T00:00:00.000Z,
    "posteCode": "J02924",
  },
  {
    "_id": Any<ObjectId>,
    "dateVersement": 2017-04-27T00:00:00.000Z,
    "montantAPayer": 1791,
    "montantPaye": 1791,
    "periodeDebut": 2017-04-01T00:00:00.000Z,
    "periodeFin": 2017-06-30T00:00:00.000Z,
    "posteCode": "J02924",
  },
  {
    "_id": Any<ObjectId>,
    "dateVersement": 2017-07-11T00:00:00.000Z,
    "montantAPayer": 1791,
    "montantPaye": 1791,
    "periodeDebut": 2017-07-01T00:00:00.000Z,
    "periodeFin": 2017-09-30T00:00:00.000Z,
    "posteCode": "J02924",
  },
  {
    "_id": Any<ObjectId>,
    "dateVersement": 2017-10-19T00:00:00.000Z,
    "montantAPayer": 1791,
    "montantPaye": 1791,
    "periodeDebut": 2017-10-01T00:00:00.000Z,
    "periodeFin": 2017-12-31T00:00:00.000Z,
    "posteCode": "J02924",
  },
]
`;
